src/main/java/graphql/schema/GraphQLInterfaceType.java

```diff
@@ -401,21 +401,25 @@ public Builder clearDirectives() {
             return this;
         }
 
-        public Builder withInterface(GraphQLInterfaceType interfaceType) {
-            assertNotNull(interfaceType, () -> "interfaceType can't be null");
-            this.interfaces.put(interfaceType.getName(), interfaceType);
-            return this;
-        }
-
-        public Builder replaceInterfaces(List<GraphQLInterfaceType> interfaces) {
+        public Builder replaceInterfaces(List<? extends GraphQLNamedOutputType> interfaces) {
```

Comment:
This is an API breaking change - albeit  a low likelyhood one - but a breaks a break so if we can NOT do it - let not do it.

This can be rewritten like this to avoid the breaking change

```
        public Builder replaceInterfaces(List<GraphQLInterfaceType> interfaces) {
            return replaceInterfacesOrReferences(interfaces);
        }

        public Builder replaceInterfacesOrReferences(List<? extends GraphQLNamedOutputType> interfacesOrReferences) {

```

and then up above you can change the impl to be

```
    public GraphQLInterfaceType withNewChildren(SchemaElementChildrenContainer newChildren) {
        return transform(builder ->
                builder.replaceDirectives(newChildren.getChildren(CHILD_DIRECTIVES))
                        .replaceFields(newChildren.getChildren(CHILD_FIELD_DEFINITIONS))
                        .replaceInterfacesOrReferences(newChildren.getChildren(CHILD_INTERFACES))
        );
    }

```